Arrow.swift

```diff
@@ -57,6 +57,29 @@ func parseType<T>(inout left:T?,right:AnyObject?) {
     }
 }
 
+// Support Array of plain Types
+
+func parseArray<T>(inout left: [T]?, right: AnyObject?) {
+    if let a = right as? [AnyObject] {
+        let tmp: [T] = a.flatMap { var t: T?; parseType(&t, right: $0); return t }
+        if tmp.count == a.count {
```

Comment:
I decided to make sure that the parsed array has the same count as the original one, and fail otherwise.
e.g. `[Int] <-- ["1", "X", "2", "Y"]` will fail, although it would be `[1, 2]` without this check.